pika/connection.py

```diff
@@ -1667,40 +1714,30 @@ def _read_frame(self):
         """
         return frame.decode_frame(self._frame_buffer)
 
-    def _reject_out_of_band_delivery(self, channel_number, delivery_tag):
```

Comment:
Just to reiterate, I think there is a case where you could have a pile of frames on the stack that are unprocessed when a Channel.Close is emitted. My thinking is that issuing the close should be similar to Basic.Cancel. I guess one could make the argument that both should not be respected until their respective Ok frames are received though.